### PR TITLE
add explicit document open API

### PR DESCRIPTION
**Pull request by kevinushey**
### Intent

Currently, there is no synchronous way for a user to open a document using RStudio. In particular, it is challenging to:

1. Create a new document in the IDE;
2. Receive an ID for that newly-created document;
3. Interact with that document.

This PR introduces an explicit API for opening a document on the user's filesystem. If that file does not already exist, then it will be created first.

### Approach

Nothing notable; implementation is mostly straightforward.

### QA Notes

Try opening various documents with `.rs.api.documentOpen()`, and see what happens when you specify "normal" paths, and perhaps also "weird" paths.

### CODE REVIEW

**Comment by kevinushey** [comment]
I've also updated `documentNew()` -- it is now synchronous, and also returns a document ID (so that newly-created documents are now much easier to interact with)

File: src/gwt/src/org/rstudio/studio/client/workbench/views/console/shell/events/SuppressNextShellFocusEvent.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ */
+package org.rstudio.studio.client.workbench.views.console.shell.events;
+
+import com.google.gwt.core.client.JavaScriptObject;
+import com.google.gwt.event.shared.EventHandler;
+import com.google.gwt.event.shared.GwtEvent;
+
+public class SuppressNextShellFocusEvent extends GwtEvent<SuppressNextShellFocusEvent.Handler>
+{
+   public static class Data extends JavaScriptObject
```
Comment (jmcphers) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/console/shell/events/SuppressNextShellFocusEvent.java:23]: Perhaps I'm missing something but I don't see the point of this class?

File: src/cpp/r/R/Api.R [hunk 2]
```diff
@@ -641,26 +643,53 @@
 
 .rs.addApiFunction("documentNew", function(type,
                                            code,
-                                           row = 0,
-                                           column = 0,
+                                           row = -1L,
+                                           column = -1L,
```
Comment (jmcphers) [inline src/cpp/r/R/Api.R:647]: Do we need to worry about existing API usage that calls documentNew w/o specifying row and column? (Will it still do the same thing?)

File: src/gwt/src/org/rstudio/studio/client/workbench/views/console/shell/events/SuppressNextShellFocusEvent.java (hunk 1 above)
Comment (kevinushey) [reply src/gwt/src/org/rstudio/studio/client/workbench/views/console/shell/events/SuppressNextShellFocusEvent.java:23]: Sorry, it's unneeded -- it was just leftover boilerplate from the generated template I had.

**Pull request merged**